File: tests/fixtures/v2-dynamic-code-flow/positive/dynamic-code-flow.js
function direct(req) {
  return eval(req.body.code)
}

function propagated(req) {
  const code = req.body.code
  const alias = code
  return eval(alias)
}

function restricted(req, vm) {
  return vm.runInNewContext(req.body.restricted, Object.create(null))
}

File: tests/fixtures/v2-dynamic-code-flow/positive/DynamicCodeFlow.java
class DynamicCodeFlow {
    Object direct(HttpServletRequest request) throws Exception {
        return new ScriptEngineManager()
            .getEngineByName(request.getParameter("language"))
            .eval(request.getParameter("code"));
    }

    Object propagated(HttpServletRequest request) throws Exception {
        String language = request.getParameter("language");
        String code = request.getParameter("code");
        String alias = code;
        return new ScriptEngineManager().getEngineByName(language).eval(alias);
    }

    Object restricted(HttpServletRequest request) throws Exception {
        return new ScriptEngineManager()
            .getEngineByName("javascript")
            .eval(request.getParameter("restricted"));
    }
}
